components/style/matching.rs

```diff
@@ -769,22 +769,27 @@ pub trait MatchMethods : TElement {
                 //
```

Comment:
While we're here, I think it would be nicer to change the contract of this function to only be called in the non-initial case, and stop passing an Option as old_style (we can do the same for existing_style_for_restyle_damage). That would reduce the complexity of the cases here.